paddle/fluid/framework/selected_rows.cc

```diff
@@ -120,66 +120,86 @@ bool SelectedRows::HasKey(int64_t key) const {
                                                                    : true;
 }
 
-std::vector<std::pair<int64_t, int64_t>> SelectedRows::Get(
-    const std::vector<int64_t>& keys, framework::Tensor* value) const {
+int64_t SelectedRows::AutoIndex(int64_t key) const {
+  pthread_rwlock_rdlock(&rwlock_.get()->lock);
+  auto iter = id_to_index_.find(key);
+  bool has_key = iter != id_to_index_.end();
+  if (!has_key) {
+    pthread_rwlock_unlock(&rwlock_.get()->lock);
+    PADDLE_THROW("Id %s not in table", key);
+  }
+  auto ret = iter->second;
+  pthread_rwlock_unlock(&rwlock_.get()->lock);
+  return ret;
+}
+
+int64_t SelectedRows::AutoGrownIndex(int64_t key) {
+  pthread_rwlock_rdlock(&rwlock_.get()->lock);
+  auto iter = id_to_index_.find(key);
+  if (iter == id_to_index_.end()) {
+    pthread_rwlock_unlock(&rwlock_.get()->lock);
+    pthread_rwlock_wrlock(&rwlock_.get()->lock);
+    PADDLE_ENFORCE(id_to_index_.size() == rows_.size(),
```

Comment:
Maybe write out the size in the message is better.

Reply:
done